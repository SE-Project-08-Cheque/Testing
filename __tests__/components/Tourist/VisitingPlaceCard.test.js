import React from "react";
import ReactDOM from "react-dom";
import VisitingPlaceCard from "../../../components/Tourist/VisitingPlaceCard";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders card without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<VisitingPlaceCard methods={[]} />, div);
});

it("renders visiting places card correctly", () => {
  const { getByTestId,getAllByTestId } = render(
    <VisitingPlaceCard
      placeName="Hikkaduwa"
      imageUrl="./1.png"
      distance="23.6"
      methods={["bus","van"]}
      timeToReach="20"
    />
  );
  expect(getByTestId("reachTime")).toHaveTextContent("20");
  expect(getByTestId("travel_distance")).toHaveTextContent("23.6");
  expect(getByTestId("travel_place")).toHaveTextContent("Hikkaduwa");
  getAllByTestId('travel_method').map((value, index)=>{
    expect(value).toHaveTextContent(["bus","van","car"][index]);
  });
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <VisitingPlaceCard
      placeName="Hikkaduwa"
      imageUrl="./1.png"
      distance="23.6"
      methods={["bus", "van", "car"]}
      timeToReach="20"
    />
  );

  expect(tree).toMatchSnapshot();
});
