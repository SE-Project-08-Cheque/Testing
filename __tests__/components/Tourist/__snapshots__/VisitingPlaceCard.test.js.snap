// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-zn4p7q"
>
  <div
    className="css-19fmqat"
  >
    <div
      className="css-1txosv4"
    >
      <div
        className="css-gmuwbf"
      >
        <img
          alt="place name"
          className="chakra-image__placeholder css-0"
        />
      </div>
    </div>
    <div
      className="css-1y4dbot"
    >
      <div
        className="css-n16x53"
      >
        <div
          className="css-x1sij0"
        >
          <span
            className="chakra-badge css-637omx"
          >
            New
          </span>
        </div>
        <h4
          className="css-1mwm0bw"
          data-testId="travel_place"
        >
          <p
            className="chakra-text css-1tapsqs"
          >
            Hikkaduwa
          </p>
        </h4>
        <div
          className="css-0"
          data-testId="travel_distance"
        >
          23.6
           km from the hotel
          <span
            className="css-bgvdb6"
          />
        </div>
        <div
          className="css-0"
          data-testId="reachTime"
        >
          20
           min journey
          <span
            className="css-bgvdb6"
          />
        </div>
        <div
          className="css-15yujhg"
          data-testId="travel_method"
        >
          <p
            className="chakra-text css-g5hezf"
          >
            Travel by :
          </p>
          <span
            className="chakra-badge css-1e8ypgz"
          >
            bus
          </span>
          <span
            className="chakra-badge css-1e8ypgz"
          >
            van
          </span>
          <span
            className="chakra-badge css-1e8ypgz"
          >
            car
          </span>
        </div>
      </div>
    </div>
    <div
      className="css-dfiqwb"
    >
      <button
        className="chakra-button css-oqfhsh"
        onClick={[Function]}
        type="button"
      >
        Details
        <span
          className="chakra-button__icon css-1oeb2oe"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
